plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/SwitchGroup/hooks.ts

```diff
@@ -0,0 +1,127 @@
+import { array_move } from "@web/extensions/sidebar/utils";
+import { useCallback, useEffect, useState } from "react";
+
+import { groupItem, SwitchGroup } from "../../types";
+
+export default (value: SwitchGroup) => {
+  const [switchGroupObj, updateGroups] = useState<SwitchGroup>(value);
+  const [groupsTitle, setGroupsTitle] = useState(value.title);
+  const [currentGroup, setCurrentGroup] = useState<groupItem>(value.groups[0]);
+  const [modifiedGroups, updateModifiedGroups] = useState<SwitchGroup | undefined>();
+
+  //initialize the helper array (modifiedGroups)
+  useEffect(() => {
+    updateModifiedGroups({ type: "switchGroup", title: "Temp", groups: [] });
+  }, []);
+
+  //add empty item each time we press on add item
+  const handAddItem = useCallback(() => {
+    updateModifiedGroups(switchGroup => {
+      if (!switchGroup) return;
+      switchGroup.groups.length == 0
+        ? switchGroup.groups.push({ id: 0, group: currentGroup.group, title: currentGroup.title })
+        : switchGroup.groups.push({ id: 0, group: "", title: "" });
```

Comment:
Here you aren't using useState correctly. You don't want to directly mutate the `switchGroup`, but instead create a new variable that you will pass in the return as the new value.
ex
```
const newGroupsArray = [];
newGroupsArray.push(laksjdflksjd);
return { ...switchGroup, groups: newGroupsArray }
```

Reply:
Also, 2 more things:
- If you are going to give IDs, you should make them unique. (though title might be enough??)
-  I don't quite understand the conditional logic. Isn't `currentGroup` for the non-editor UI? Shouldn't this be only pushing a new group to the end of the array and that is it?